pkg/webhook/image_swapper.go

```diff
@@ -204,48 +204,50 @@ func (p *ImageSwapper) Mutate(ctx context.Context, ar *kwhmodel.AdmissionReview,
 
 			targetImage := p.targetName(srcRef)
 
-			copyFn := func() {
-				// Avoid unnecessary copying by ending early. For images such as :latest we adhere to the
-				// image pull policy.
-				if p.registryClient.ImageExists(targetImage) && container.ImagePullPolicy != corev1.PullAlways {
-					return
+			copyFn := func(targetImage string, container corev1.Container) func() {
```

Comment:
While this may be a quick fix and work very well, wouldn't a cleaner solution be to have an object for each "ImageCopier" with a `.CopyFn()`.  This would avoid any confusion caused by mutating variables and their scope(s).